packages/core/src/main/client/ClientDatabaseRepository.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {CRUDEngine} from '@wireapp/store-engine/dist/commonjs/engine/index';
+import {RegisteredClient} from '@wireapp/api-client/dist/commonjs/client/index';
+import {CryptographyService} from '../cryptography/root';
+import {MetaClient} from './ClientService';
+
+export enum DatabaseStores {
+  CLIENTS = 'clients',
+}
+
+class ClientDatabaseRepository {
+  public static readonly STORES = DatabaseStores;
+
+  public static KEYS = {
+    LOCAL_IDENTITY: 'local_identity',
+  };
+
+  constructor(private storeEngine: CRUDEngine) {}
+
+  public getLocalIdentityClient(): Promise<MetaClient> {
+    return this.getClient(ClientDatabaseRepository.KEYS.LOCAL_IDENTITY);
+  }
+
+  public getClient(sessionId: string): Promise<MetaClient> {
+    return this.storeEngine.read<MetaClient>(ClientDatabaseRepository.STORES.CLIENTS, sessionId);
+  }
+
+  public deleteLocalIdentityClient(): Promise<string> {
+    return this.storeEngine.delete(
+      ClientDatabaseRepository.STORES.CLIENTS,
+      ClientDatabaseRepository.KEYS.LOCAL_IDENTITY
+    );
+  }
+
+  public deleteClient(sessionId: string): Promise<string> {
+    return this.storeEngine.delete(ClientDatabaseRepository.STORES.CLIENTS, sessionId);
+  }
+
+  public createClientList(userId: string, clientList: RegisteredClient[]): Promise<MetaClient[]> {
+    const createClientTasks: Promise<MetaClient>[] = [];
+    for (const client of clientList) {
+      createClientTasks.push(this.createClient(userId, client));
+    }
+    return Promise.all(createClientTasks);
+  }
+
+  public createLocalIdentityClient(client: RegisteredClient): Promise<MetaClient> {
+    const transformedClient = this.transformLocalIdentityClient(client);
+    return this.storeEngine
+      .create(ClientDatabaseRepository.STORES.CLIENTS, ClientDatabaseRepository.KEYS.LOCAL_IDENTITY, transformedClient)
+      .then(() => transformedClient);
+  }
+
+  public updateLocalIdentityClient(client: RegisteredClient): Promise<MetaClient> {
```

Comment:
It is not really a local identity. We just named the database key at some point. Let's name it `updateLocalClient` instead. And make use of this name everywhere.